src/bip44.c

```diff
@@ -158,6 +184,19 @@ bool bip44_isValidStakingKeyPath(const bip44_path_t* pathSpec)
 	return (bip44_getAddressValue(pathSpec) == 0);
 }
 
+// Pool cold keys
+bool bip44_isValidPoolColdKeyPath(const bip44_path_t* pathSpec)
+{
+	if (!bip44_hasValidCardanoPoolColdKeyPrefix(pathSpec)) return false;
+
+#define CHECK(cond) if (!(cond)) return false
+	CHECK(pathSpec->length == POOL_COLD_KEY_PATH_LENGTH);
+	CHECK(pathSpec->path[BIP44_I_POOL_COLD_KEY_USECASE] == 0);
```

Comment:
Imho asserting in this case would make for a somewhat weird behavior as the concern of the function is to return true if it's a valid cold key path, otherwise false, so asserting for a certain subclass of paths based solely on their length does not feel right to me. I understand the concern that here we are potentially accessing something beyond the boundary, but it's obvious from the surrounding code that can't really happen